### ISSUE TITLE
Slower factorization

### ISSUE THREAD

**Issue by mohamed-barakat** [oscar-system/Singular.jl#274]
Now that #270 has been resolved step 8 of [this notebook](https://github.com/homalg-project/CapHomalgNotebooks/blob/master/uMPS(2%2C2%2C4)_0dim.ipynb) is running very slow.

It used to take less than 15 seconds before switching to `Singular_jll`. I am quite sure the problem is a significantly slower computation of radicals due to much slower factorization.

**Comment by fingolfin** [comment]
Step 8 is `im = ConstructibleImage( ϕ )` ? Can you perhaps reduce that example to an invocation of Singular and/or Singular.jl (assuming you use that)?

**Comment by mohamed-barakat** [comment]
> Step 8 is `im = ConstructibleImage( ϕ )` ?

Yes

> Can you perhaps reduce that example to an invocation of Singular and/or Singular.jl (assuming you use that)?

Yes, I will try to isolate it to a Singular call.

**Comment by mohamed-barakat** [comment]
The last line of the attached file

```
matrix homalg_variable_758 = RadicalSubobject(homalg_variable_752)
```

runs in less than a second with the standalone Singular (https://github.com/Singular/Singular/commit/d51aaf5d031f0fee4b8bedcebd3047a50c8cebe1)
```
Singular for x86_64-Darwin version 4.1.3 (4132, 64 bit) Sep  3 2020 15:24:11
with
	GMP(6.2.0),NTL(11.4.3),FLINT(2.6.3),
	omalloc,static readline(8),Plural,DBM,
	dynamic modules,dynamic p_Procs,OM_NDEBUG,SING_NDEBUG,
	CC:OPTIMIZE,GenericAdd,TableMult,no invTable,
	eigenvalues,Gauss-Manin system,random=1602677554
```

However, using the Singular within Singular.jl (or the binary `~/.julia/artifacts/10a4a3842417ad1c768c82c3bcc91c1a93576c37/bin/Singular`)

the computation time increases significantly.

[SlowRadical.txt](https://github.com/oscar-system/Singular.jl/files/5378053/SlowRadical.txt)

**Comment by mohamed-barakat** [comment]
For the record: This is the reason why the notebook `uMPS(2,2,4)_0dim` is running out of time [here](https://oscar-system.github.io/oscar-ci-status/index.html).

**Comment by fingolfin** [comment]
So Dan just told me that Flint can handle this factorization very quickly. The problem is to hook it up into Singular, and also to adjust the Singular test suite (it works a bit like the on in GAP, i.e., it compares printed output of operations; but using a different factorization algorithm may return factors in a different order, which then implies that test results may look differently. But all that sounds surmountable to me (in GAP we do similar test adjustments regularly with no major issue).

**Comment by hannes14** [comment]
Singular is now adapted to use Flints multivariate factorization (if available). To use this in Singular.jl/Oscar.jl we need
a libflint_jll with the new, not yet released Flint 2.7.0, and a libsingular_jll which uses this libflint_jll.

**Comment by fingolfin** [comment]
With PR #321 this runs in less than 0.1 seconds for me.
